model/monitorCache/implementation/DeliveryMonitoringData.php

```diff
@@ -256,38 +257,9 @@ private function updateExtraTime()
             $timer->setStorage(new QtiTimeStorage($this->deliveryExecution->getIdentifier(), $this->deliveryExecution->getUserIdentifier()));
             $timer->load();
         }
-        $maxTimeSeconds = null;
-
-        if ($timeLimits = $testSession->getAssessmentTest()->getTimeLimits()) {
-            $timeLimits = $testSession->getAssessmentTest()->getTimeLimits();
-            $maxTimeSeconds = $timeLimits->hasMaxTime()
-                ? $timeLimits->getMaxTime()->getSeconds(true)
-                : null;
-        }
-
-        if ($testPart = $testSession->getCurrentTestPart()) {
-            if ($testSessionLimits = $testPart->getTimeLimits()) {
-                $maxTimeSeconds = $testSessionLimits->hasMaxTime()
-                    ? $testSessionLimits->getMaxTime()->getSeconds(true)
-                    : $maxTimeSeconds;
-            }
-        }
-
-        if ($section = $testSession->getCurrentAssessmentSection()) {
-            if ($testSessionLimits = $section->getTimeLimits()) {
-                $maxTimeSeconds = $testSessionLimits->hasMaxTime()
-                    ? $testSessionLimits->getMaxTime()->getSeconds(true)
-                    : $maxTimeSeconds;
-            }
-        }
-
-        if ($test = $testSession->getCurrentAssessmentItemSession()) {
-            if ($testSessionLimits = $test->getTimeLimits()) {
-                $maxTimeSeconds = $testSessionLimits->hasMaxTime()
-                    ? $testSessionLimits->getMaxTime()->getSeconds(true)
-                    : $maxTimeSeconds;
-            }
-        }
+        /** @var QtiRunnerService $qtiRunnerService */
+        $qtiRunnerService = $this->getServiceLocator()->get(QtiRunnerService::SERVICE_ID);
+        $maxTimeSeconds = $qtiRunnerService->getTimeLimitsFromSession($testSession);
```

Comment:
This function exists since version `10.18.1` of taoQtiTest. Please update dependency.